FusionIIIT/applications/counselling_cell/models.py

```diff
@@ -1,3 +1,119 @@
 from django.db import models
-
+from applications.academic_information.models import Student
+from django.contrib.auth.models import User
+from applications.globals.models import Faculty
 # Create your models here.
+
+class CounsellingCellConstants :
+    STUDENT_POSTIONS= (
+        ('student_guide', 'Student Guide'),
+        ('student_coordinator', 'Student Coordinator'),
+    )
+    FACULTY_POSTIONS= (
+        ('faculty_counsellor', 'Faculty Counsellor'),
+        ('counsellor', 'Counsellor'),
+    )
+    
+    ISSUE_STATUS= (
+        ('status_unresolved', 'Unresolved'),
+        ('status_resolved', 'Resolved'),
+        ('status_inprogress', 'InProgress'),
+    )
+
+    MEETING_STATUS = (
+        ('status_accepted',"Accepted"),
+        ('status_pending','Pending')
+    )
+
+class FacultyCounsellingTeam(models.Model):
+    faculty = models.ForeignKey(Faculty, on_delete=models.CASCADE)
+    faculty_position = models.CharField(max_length=50,choices=CounsellingCellConstants.FACULTY_POSTIONS,required=True)
+
+    class Meta:
+        unique_together = (('faculty', 'faculty_position'))
+
+    def __str__(self):
+        return f"{self.faculty} - {self.faculty_position}"
+
+class StudentCounsellingTeam(models.Model):
+    student = models.ForeignKey(Student, on_delete=models.CASCADE)
+    student_position = models.CharField(max_length=50,choices=CounsellingCellConstants.STUDENT_POSTIONS,required=True)
+
+    class Meta:
+        unique_together = (('student_id', 'student_position'))
+
+    def __str__(self):
+        return f"{self.student} - {self.student_position}"
+
+class StudentCounseliingInfo(models.Model):
+    faculty = models.ForeignKey(FacultyCounsellingInfo ,on_delete=models.CASCADE)
+    student_guide = models.ForeignKey(StudentCounsellingInfo,on_delete=models.CASCADE)
+    student = models.ForeignKey(Student,on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = (('faculty', 'student_guide','student'))
+
+    def __str__(self):
+        return f"{self.faculty} - {self.student_guide} - {self.student}"
+
+class CounsellingIssueCategory(models.Model):
+    category = models.CharField(max_length=40)
```

Comment:
An `category_id` char field would be good here, like the two elements you have in tuples above. 

Ex. for Academic issue category, `category_id` can be given a value of `academics` and `category` can be given a value of `Academinc`.